### ISSUE TITLE
using alias does not work with private type

### ISSUE THREAD

**Issue by YairHalberstadt** [dotnet/roslyn#42568]
**Version Used**: master as of 9th march

**Steps to Reproduce**:

```csharp
using I = Outer.Inner;

public class Outer
{
    
    public void M() {
        var x = new I();
    }
    
    private class Inner
    {
    }
}
```

**Expected Behavior**: No error

**Actual Behavior**:

error CS0122: 'Outer.Inner' is inaccessible due to its protection level

I believe this is a bug, because the [Spec](https://github.com/dotnet/csharplang/blob/master/spec/namespaces.md#using-alias-directives) makes no mention of such a restriction.

**Comment by YairHalberstadt** [comment]
According to @AlekseyTs this is to spec: See his comments at https://github.com/dotnet/csharplang/pull/3299.
